Feature/child work packages

ulferts opened the pull request:
Implements creation of child work packages from each work package's relation tab as specified by:

https://community.openproject.org/work_packages/22178/activity

The functionality makes use of the `parent_id` field that is defined on the work package resource. IMO this field violates the API design. I have changed the front end application to only use the linked `parent` resource but did not remove the `parent_id` field as I don't think that it would be good practice to do that in a patch level release. The field should be removed on dev though.

ulferts committed:
add parentId field to schema

This is somewhat debateable. Having a parentId field is wrong in itself. It should be a linked resource. But as the field is also present in the resource, it also needs to be part of the schema for consistency sake.

ulferts committed:
fix addChild path in work package resource

ulferts committed:
implement work package child creation

ulferts committed:
enable child work package creation in split view

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://ulferts-feature-child-work-packages.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://ulferts-feature-child-work-packages.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

ulferts committed:
use linked parent resource instead of parentId

The parentId field should be removed. But we shouldn't do it in a
minor version.

oliverguenther commented:
LGTM. :+1:  Two questions (not necessarility related):
1. It might make sense to disable the create child button for the Milestone type, since I cannot create descendants there.
   ![bildschirmfoto 2015-12-11 um 08 21 18](https://cloud.githubusercontent.com/assets/459462/11738164/7af65784-9fe0-11e5-8835-f63aba37f29b.png)
2. Why are all other 'Add X' relation disabled?

Even though we might address 1. now, I'm merging this for 5.0.7.
